core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -418,6 +418,32 @@ class DAGScheduler(
     updateJobIdStageIdMapsList(List(stage))
   }
 
+  /** Clear references to the given stage in stored data structures */
+  private def removeStage(stageId: Int) {
```

Comment:
By moving this you've changed the scope in which removeStage is visible, potentially opening up improper calls to removeStage from elsewhere than intended in the DAGScheduler.